tests/php/modules/contact-form/test-class.grunion-contact-form.php

```diff
@@ -867,28 +894,49 @@ public function assertValidFieldMultiField( $html, $attributes ) {
 
 			$this->assertEquals( $select->getAttribute( 'class' ), 'select '. $attributes['class'], ' select class does not match expected' );
 
-			// First Option
-			$option = $this->getFirstElement( $select, 'option' );
-			$this->assertEquals( $option->getAttribute( 'value' ), $attributes['values'][0], 'Input value doesn\'t match' );
-			$this->assertEquals( $option->getAttribute( 'selected' ), 'selected', 'Input is not selected' );
-			$this->assertEquals( $option->nodeValue, $attributes['options'][0], 'Input does not match the option' );
-
+			// Options.
+			$options = $select->getElementsByTagName( 'option' );
+			$n       = is_array( $options ) ? count( $options ) : $options->length;
+			$this->assertEquals( $n, count( $attributes['options'] ), 'Number of inputs doesn\'t match number of options' );
+			$this->assertEquals( $n, count( $attributes['values'] ), 'Number of inputs doesn\'t match number of values' );
+			for ( $i = 0; $i < $n; $i++ ) {
+				$option = is_array( $options ) ? $options[ $i ] : $options->item( $i );
+				$this->assertEquals( $option->getAttribute( 'value' ), $attributes['values'][ $i ], 'Input value doesn\'t match' );
+				if ( 0 === $i ) {
+					$this->assertEquals( $option->getAttribute( 'selected' ), 'selected', 'Input is not selected' );
+				} else {
+					$this->assertNotEquals( $option->getAttribute( 'selected' ), 'selected', 'Input is selected' );
+				}
+				//phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$this->assertEquals( $option->nodeValue, $attributes['options'][ $i ], 'Input does not match the option' );
+			}
 		} else {
 			$this->assertEquals( $label->getAttribute( 'class' ), 'grunion-field-label', 'label class doesn\'t match' );
-			// Radio and Checkboxes
-			$second_label = $this->getFirstElement( $wrapperDiv, 'label', 1 );
-			$this->assertEquals( $second_label->nodeValue, ' ' . $attributes['options'][0] ); // extra space added for a padding
-
-			$input = $this->getFirstElement( $second_label, 'input' );
-			$this->assertEquals( $input->getAttribute( 'type' ), $attributes['input_type'], 'Type doesn\'t match' );
-			if (  $attributes['input_type'] === 'radio' ) {
-				$this->assertEquals( $input->getAttribute( 'name' ), $attributes['id'], 'Input name doesn\'t match' );
-			} else {
-				$this->assertEquals( $input->getAttribute( 'name' ), $attributes['id'] . '[]', 'Input name doesn\'t match' );
+			// Radio and Checkboxes.
+			$labels = $wrapper_div->getElementsByTagName( 'label' );
+			$n      = is_array( $labels ) ? count( $labels ) - 1 : $labels->length - 1;
+			$this->assertEquals( $n, count( $attributes['options'] ), 'Number of inputs doesn\'t match number of options' );
+			$this->assertEquals( $n, count( $attributes['values'] ), 'Number of inputs doesn\'t match number of values' );
+			for ( $i = 1; $i < $n; $i++ ) {
```

Comment:
Why does the loop start at 1? We're using `$i+1` for the indexes below, so I'm not sure why we need to start this loop at 1.

Reply:
Good catch, thanks!

I probably got confused between doing +1 for the $labels accesses versus -1 for the accesses of the options and values.